frontend/src/HexView.svelte

```diff
@@ -167,32 +160,43 @@
   }
   $: childRangesPromise = calculateRanges($selectedResource, dataPromise);
 
-  function getRangeInfo(index, childRanges) {
+  function getRangeInfo(T, childRanges) {
```

Comment:
I did a while back, and there were some issues with it. It may be worth revisiting later, but in the meantime, the current method is definitely sufficiently fast.